src/org/scratchOrgSettingsGenerator.ts

```diff
@@ -30,23 +29,56 @@ export enum RequestStatus {
 
 const breakPolling = ['Succeeded', 'SucceededPartial', 'Failed', 'Canceled'];
 
-interface ObjectToBusinessProcessPicklist {
-  [key: string]: {
-    fullName: string;
-    default?: boolean;
+export interface SettingType {
+  members: string[];
+  name: string;
+}
+
+export interface PackageFile {
+  '@': {
+    xmlns: string;
   };
+  types: SettingType[];
+  version: string;
 }
 
-export interface BusinessProcessFileContent extends JsonMap {
-  fullName: string;
-  isActive: boolean;
-  values: [
-    {
-      fullName: string;
-      default?: boolean;
+export const writePackageFile = ({
+  allRecordTypes = [],
```

Comment:
`allRecordTypes` and `allBusinessProcesses` are defaulted to `[]` but marked optional. Usually you just to define a default, or as optional not both. If you define everything with a default you could remove some of the logic below, like `if(settingData)`, since you knew it would always be populated, even if was the default of `[]`

Reply:
Actually the way it's written it makes perfect sense if you read the test and the way conditionals are being used, as far I know it breaks no rules and allows for:

1. A default value allows to write a very simple conditional like so: `if (allRecordTypes.length > 0)` which saves us from asserting the possible `undefined` value.
2. An optional argument allows to write tests without the need to pass in empty arrays like so: `createObjectFileContent({ apiVersion: '54' });`
Unless there is a good reason not to have them written this way I want to keep it this way.

Reply:
1. it's only not breaking rules because the argument into the method is an object, if you broke them apart, so that the method required 4 arguments, it would throw an error 
![image](https://user-images.githubusercontent.com/23369140/176463389-9c587346-2afb-4b85-8dd4-577cb973739e.png)

2. if they all had defaults, you could remove the check, because `Object.keys` or however you were iterating over them would continue to work, it would loop 0 times

maybe it's a valid use case of an optional argument having a default, or they should just be optional. I think each option should follow the same pattern. It seems like you could simplify some of the logic below by changing some of this